### PR TITLE
Add new command /rmall

### PR DESCRIPTION
**Pull request by DeathlyPlays**

### CODE REVIEW

File: spacialgaze-plugins/misc.js [hunk 1]
```diff
@@ -433,4 +433,21 @@ exports.commands = {
...
+		if (!target) return this.sendReply("/rmall [message] - Sends a pm to all users in the room.");
+		target = target.replace(/<(?:.|\n)*?>/gm, '');
+
+		let pmName = ' SG Server';
+
+		for (let i in room.users) {
+			let message = '|pm|' + pmName + '|' + room.users[i].getIdentity() + '| ' + target;
+			room.users[i].send(message);
+		}
+		this.privateModCommand('(' + Chat.escapeHTML(user.name) + ' mass PMd: ' + target + ')');
```
Comment (Lord-Haji) [inline spacialgaze-plugins/misc.js]: It Should be room PM'd

File: spacialgaze-plugins/misc.js [hunk 2]
```diff
@@ -433,4 +433,21 @@ exports.commands = {
 		Db.disabledScrolls.remove(target);
 	},
 	enableintroscrollhelp: ["/enableintroscroll [room] - Enables scroll bar preset in the room's roomintro."],
+
+	pmroom: 'rmall',
+	roompm: 'rmall',
+	rmall: function (target, room, user) {
+		if (!this.can('declare', null, room)) return this.errorReply("/rmall - Access denied.");
+		if (room.id === 'lobby') return this.errorReply("This command cannot be used in Lobby.");
```
Comment (Lord-Haji) [inline spacialgaze-plugins/misc.js]: I don't see a point to not to be allowed in lobby

**Review by Lord-Haji — CHANGES_REQUESTED** [review]

File: spacialgaze-plugins/misc.js [hunk 3]
```diff
@@ -433,4 +433,21 @@ exports.commands = {
...
+		if (!target) return this.sendReply("/rmall [message] - Sends a pm to all users in the room.");
+		target = target.replace(/<(?:.|\n)*?>/gm, '');
+
+		let pmName = ' SG Server';
```
Comment (Lord-Haji) [inline spacialgaze-plugins/misc.js]: No point of the extra space before SG Server
And I think it should be the room name instead SG Server coz ppl might confuse it as a global PM

**Review by Lord-Haji — CHANGES_REQUESTED** [review]

File: spacialgaze-plugins/misc.js (hunk 3 above)
Comment (DeathlyPlays) [reply spacialgaze-plugins/misc.js]: To make it unPMable this is best

File: spacialgaze-plugins/misc.js (hunk 2 above)
Comment (DeathlyPlays) [reply spacialgaze-plugins/misc.js]: Spam...

File: spacialgaze-plugins/misc.js (hunk 1 above)
Comment (DeathlyPlays) [reply spacialgaze-plugins/misc.js]: Mass PM'ed is fine.

Comment (Lord-Haji) [reply spacialgaze-plugins/misc.js]: Mass PM'ed is used in /pmall to so this is not fine

Comment (Lord-Haji) [reply spacialgaze-plugins/misc.js]: Mass PM'ed is used in /pmall to so this is not fine

File: spacialgaze-plugins/misc.js (hunk 2 above)
Comment (Lord-Haji) [reply spacialgaze-plugins/misc.js]: It needs roomowner and why will a room owner spam in lobby
P.S SG doesn't even have lobby staff

File: spacialgaze-plugins/misc.js (hunk 3 above)
Comment (Lord-Haji) [reply spacialgaze-plugins/misc.js]: Huh? Didn't get u

File: spacialgaze-plugins/misc.js (hunk 1 above)
Comment (DeathlyPlays) [reply spacialgaze-plugins/misc.js]: Who cares....

**Comment by DeathlyPlays** [comment]
@HoeenCoder whats your jurisdiction on this so I can close/push this to SG?

**Comment by DeathlyPlays** [comment]
You've seen it works lmao

**Comment by DeathlyPlays** [comment]
And because @Lord-Haji is pissing me off right now about stupid shit that's irrelevant

File: spacialgaze-plugins/misc.js (hunk 3 above)
Comment (HoeenCoder) [reply spacialgaze-plugins/misc.js]: We cannot use names that I havent registered and set aside for this, please use `SG Server`

File: spacialgaze-plugins/misc.js (hunk 1 above)
Comment (HoeenCoder) [reply spacialgaze-plugins/misc.js]: `mass room PMed` would be better tbh

File: spacialgaze-plugins/misc.js (hunk 2 above)
Comment (HoeenCoder) [reply spacialgaze-plugins/misc.js]: Its fine in lobby, useful for uno and tours actually

File: spacialgaze-plugins/misc.js (hunk 3 above)
Comment (Mystifi) [reply spacialgaze-plugins/misc.js]: @Lord-Haji, the extra space is actually the user's rank. When messages show up in PMs, the protocol is like this:

`|pm|<rank><sender>|<rank><receiver>|<message>`

If the sender doesn't have a valid global rank, a space is put there and is interpreted as a regular user.

So, yes, the space is actually needed. Please learn the protocol before saying stuff like that

**Pull request merged**